### ISSUE TITLE
Flatpack installation issues

### ISSUE THREAD

**Issue by cyanit** [elementary/appcenter#1234]
## The problem
This seems like a chain of unlucky events to me, so ill describe everything below with the current/main problem seperated.

I tried to install a flatpack app (Gitkraken) which took longer than expected, I had to suspend my machine. Coming home later and waking it up again the installation seemed stuck. So I tried to cancel and restart the installation, which resulted in the app center freezing up. I restarted the machine, installed the "Monitor" task manager (which worked perfectly) and tried downloading gitkraken again, which does not work. I just tried to install another

## Problematic Behaviour
Now when clicking the install button it:
1. Show the progress bar "Waiting" for half a second
2. Fills the progress bar "Installing" to a third
3. Empties it again, now showing "Starting" for a moment
4. Returns to the initial state like I never clicked install.

**TL;DR:** Installation of previously not correctly installed flatpak apps just fails without any error.

**To reproduce** download a flatpak app, kill appcenter or restart/suspend your machine and things wont work.

### Expected behavior
Resuming the installation of an app after suspension, or in case of a failed install, show an error message with a path to the log and the option to reinstall.

### Logs
This is what running AppCenter in GDB spits out:

```
** (io.elementary.appcenter:5784): WARNING **: 17:45:46.698: FlatpakBackend.vala:705: Flatpak installation failed: Failed to read commit ea1850468dc2940df8f6544d157688a98bce3a8460a7a4fe672c0150e48ec627: No such metadata object 1fcac62d058b58fe9274ff6406868d6ad35551fa37235187c0cd243a2d6cbeef.dirtree (detail: 1)
[New Thread 0x7fffd5a16700 (LWP 6579)]
[Thread 0x7fffd5a16700 (LWP 6579) exited]
[New Thread 0x7fffd5a16700 (LWP 6584)]
[Thread 0x7fffd5a16700 (LWP 6584) exited]
```

```
** (io.elementary.appcenter:5784): WARNING **: 17:45:47.367: FlatpakBackend.vala:705: Flatpak installation failed: While trying to checkout 2fc58e3050e8fa0f8245e29ed8081f55f8520e35bb921e412df499267d56414a into /home/leo/.local/share/flatpak/runtime/org.gnome.Platform/x86_64/3.34/.2fc58e3050e8fa0f8245e29ed8081f55f8520e35bb921e412df499267d56414a-ZXLUH0: Opening content object 60e9855619c92ef3c93af35e911a6c69cb92185df6fe8e3203709eddf8484283: Couldn't find file object '60e9855619c92ef3c93af35e911a6c69cb92185df6fe8e3203709eddf8484283' (detail: 0)
```

## Platform Information
![Screenshot from 2020-03-22 17-49-05](https://user-images.githubusercontent.com/26248671/77255118-7940ad80-6c65-11ea-8ba8-3181fa3b93e0.png)

- [x] I'm using the latest released stable version of the OS and AppCenter

Thats about it, I hope that someone will find some time and a way to fix this (or in case this is just my fault for suspending the machine, id greatly appreciate any tips on fixing it) ^^

**Comment by cyanit** [comment]
Update: just saw theres an os update for the component "org.freedesktop.Platform Flatpak runtime"

unfortunally I cant install thet either

**Comment by davidmhewitt** [comment]
If you also have problems with the `flatpak` command, there is not much AppCenter can do.

Could you try running `flatpak repair --user` from the terminal?

**Comment by davidmhewitt** [comment]
I'm going to close this issue due to inactivity.

As discussed, this is not directly a problem with AppCenter, since the flatpak CLI also can't perform updates. I've suggested a possible way to repair this and we've also pushed a new version of AppCenter recently that should prevent the computer suspending if it's in the middle of a package operation. Which hopefully means it'll be harder to get into this state.
